pkg/controller/logstorage/linseed/esgateway.go

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021-2023 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linseed
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	octrl "github.com/tigera/operator/pkg/controller"
+	"github.com/tigera/operator/pkg/controller/certificatemanager"
+	lscommon "github.com/tigera/operator/pkg/controller/logstorage/common"
+	"github.com/tigera/operator/pkg/controller/utils"
+	"github.com/tigera/operator/pkg/controller/utils/imageset"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	"github.com/tigera/operator/pkg/render/logstorage/esgateway"
+	"github.com/tigera/operator/pkg/tls/certificatemanagement"
+)
+
+func (r *LinseedSubController) createESGateway(
+	ctx context.Context,
+	request octrl.Request,
+	install *operatorv1.InstallationSpec,
+	variant operatorv1.ProductVariant,
+	pullSecrets []*corev1.Secret,
+	hdler utils.ComponentHandler,
+	reqLogger logr.Logger,
+	usePSP bool,
+) error {
+	// Create a trusted bundle to pass to the render pacakge. The actual contents of this bundle don't matter - the ConfigMap
+	// itself will be managed by the Secret controller. But, we need an interface to use as an argument to render in order
+	// to configure volume mounts properly.
+	trustedBundle := certificatemanagement.CreateTrustedBundle()
```

Comment:
As it turns out, this comment is completely false - we need the _real_ trusted bundle here in order to properly populate the hash annotations on the objects that depends on this bundle.

The problem is that the bundle is created in the secrets_controller, and so this controller doesn't have access to it. 

This is further exacerbated by the fact that the trusted bundle that is written to the cluster has lost key information by the time it is stored in a ConfigMap - for example, which namespace the certificates in the bundle originated. This makes it impossible for other controllers to reach the bundle and reverse engineer it by re-loading those certificates.

I think the solution to this is for the secrets controller to write the trusted bundle with the hash annotations pre-calcualted and included in the annotations on the configmap.

Then, controllers can simply load that configmap and copy the annotations straight from the secret without ever needing to deal with the contents of the configmap itself. Would be interested in feedback on this before I implement it - probably one for @rene-dekker

Reply:
Render does need more than just the annotations - it also needs the volume mount path, and the volumes.

These can _almost_ be hard-coded, with the exception of needing to know whether or not the system certs need to be mounted.

Reply:
My latest commit includes a very quick and dirty example of what this might look like: https://github.com/tigera/operator/pull/2738/commits/c3f76da9855a853c89d0adf667b3c4347f8d2da9

Reply:
Ugh, the bundle for this controller has continuously caused issues. Glad you are getting to the root of this.